pkg/actuators/machine/reconciler.go

```diff
@@ -342,6 +351,67 @@ func (r *Reconciler) updateLoadBalancers(instance *ec2.Instance) error {
 	return nil
 }
 
+// deletePlacementGroup deletes the placement group for the machine
+func (r *Reconciler) deletePlacementGroup() error {
```

Comment:
I think it would also be good as part of this function to look for Machines that define the PG. If more than one Machine defines the PG name, we can return early as the PG does not need deleting

Reply:
Implemented!